src/runtime/pkg/oci/utils.go

```diff
@@ -1183,9 +1183,21 @@ func (a *annotationConfiguration) setUintWithCheck(f func(uint64) error) error {
 	return nil
 }
 
+func (a *annotationConfiguration) setFloat32WithCheck(f func(float32) error) error {
+	if value, ok := a.ocispec.Annotations[a.key]; ok {
+		float64Value, err := strconv.ParseFloat(value, 32)
+		if err != nil || float64Value < 0 {
+			return fmt.Errorf(errAnnotationPositiveNumericKey, a.key)
+		}
+		float32Value := float32(float64Value)
```

Comment:
While you are checking, maybe add a check for some sane maximum value?

Reply:
I think the sane maximm value here would the math.MaxFloat32.
Then whether it's bigger than the amount of CPUs present in the system is something already checked elsewhere.